src/gui/qgsrelationeditorwidget.cpp

```diff
@@ -427,27 +420,18 @@ void QgsRelationEditorWidget::updateButtons()
   mLinkFeatureButton->setEnabled( linkable );
   mDeleteFeatureButton->setEnabled( editable && selectionNotEmpty );
   mUnlinkFeatureButton->setEnabled( linkable && selectionNotEmpty );
-
-  mZoomToFeatureButton->setVisible(
-    mEditorContext.mapCanvas() && (
-      (
-        mNmRelation.isValid() &&
-        mNmRelation.referencedLayer()->geometryType() != QgsWkbTypes::NullGeometry &&
-        mNmRelation.referencedLayer()->geometryType() != QgsWkbTypes::UnknownGeometry
-      )
-      ||
-      (
-        mRelation.isValid() &&
-        mRelation.referencedLayer()->geometryType() != QgsWkbTypes::NullGeometry &&
-        mRelation.referencedLayer()->geometryType() != QgsWkbTypes::UnknownGeometry
-      )
-    )
-  );
-
   mZoomToFeatureButton->setEnabled( selectionNotEmpty );
-
   mToggleEditingButton->setChecked( editable );
   mSaveEditsButton->setEnabled( editable );
+
+  mLinkFeatureButton->setVisible( mButtonsVisibility.testFlag( QgsAttributeEditorRelation::Button::Link ) );
+  mUnlinkFeatureButton->setVisible( mButtonsVisibility.testFlag( QgsAttributeEditorRelation::Button::Unlink ) );
+  mSaveEditsButton->setVisible( mButtonsVisibility.testFlag( QgsAttributeEditorRelation::Button::SaveChildEdits ) && mToggleEditingButton->isVisible() );
```

Comment:
can't we use another condition that the button visibility?

Reply:
One possibility would be to check if the referencing layer is in the `transactionGroup` but I don't like that it iterates here since it's not really something that changed.

Another possibility would be to set the status in something like `mLayerInTransactionGroup = true` or `mLayerEditable = false` here https://github.com/qgis/QGIS/pull/38678/files#diff-52237b5949ffab9a9049c00ed3a713e1L332 and then check the member variable in the `setVisible`.

What do you think?

Reply:
yes, let's go for the member, it will make the logic more explicit.